ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/PartyLedgerEntry.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.stores.ledger
+
+import java.time.Instant
+
+import com.daml.ledger.participant.state.v1.{ParticipantId, SubmissionId}
+import com.digitalasset.ledger.api.domain.PartyDetails
+
+sealed abstract class PartyLedgerEntry() extends Product with Serializable {
+  val submissionIdOpt: Option[SubmissionId]
+  val recordTime: Instant
+}
+
+object PartyLedgerEntry {
+
+  final case class AllocationAccepted(
+      submissionIdOpt: Option[SubmissionId],
```

Comment:
@gerolf-da why is SubmissionId optional on the allocation accepted?  i believe there should always be a submissionId sent as part of the allocation request

Reply:
The `submissionId` is optional for the same reason that `commandId` is optional: only the submitting participant should see it.

Reply:
shouldn't it be consistent between the Accepted and Rejected?

Reply:
No, you should only ever see a rejection on the participant that submitted the party allocation.